github/resource_github_branch_protection.go

```diff
@@ -193,58 +203,71 @@ func resourceGithubBranchProtectionUpdate(d *schema.ResourceData, meta interface
 		return err
 	}
 
-	_, _, err = client.Repositories.UpdateBranchProtection(context.TODO(), meta.(*Organization).name, r, b, protectionRequest)
+	_, _, err = client.Repositories.UpdateBranchProtection(context.TODO(),
+		meta.(*Organization).name,
+		repoName,
+		branch,
+		protectionRequest,
+	)
 	if err != nil {
 		return err
 	}
 
 	if protectionRequest.RequiredPullRequestReviews == nil {
-		_, err = client.Repositories.RemovePullRequestReviewEnforcement(context.TODO(), meta.(*Organization).name, r, b)
+		_, err = client.Repositories.RemovePullRequestReviewEnforcement(context.TODO(),
+			meta.(*Organization).name,
+			repoName,
+			branch,
+		)
 		if err != nil {
 			return err
 		}
 	}
 
-	d.SetId(buildTwoPartID(&r, &b))
+	d.SetId(buildTwoPartID(&repoName, &branch))
 
 	return resourceGithubBranchProtectionRead(d, meta)
 }
 
 func resourceGithubBranchProtectionDelete(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*Organization).client
-	r, b, err := parseTwoPartID(d.Id())
+	repoName, branch, err := parseTwoPartID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	_, err = client.Repositories.RemoveBranchProtection(context.TODO(), meta.(*Organization).name, r, b)
+	_, err = client.Repositories.RemoveBranchProtection(context.TODO(),
+		meta.(*Organization).name,
+		repoName,
+		branch,
+	)
 	return err
 }
 
 func buildProtectionRequest(d *schema.ResourceData) (*github.ProtectionRequest, error) {
-	protectionRequest := new(github.ProtectionRequest)
+	req := &github.ProtectionRequest{
+		EnforceAdmins: d.Get("enforce_admins").(bool),
+	}
 
 	rsc, err := expandRequiredStatusChecks(d)
 	if err != nil {
 		return nil, err
 	}
-	protectionRequest.RequiredStatusChecks = rsc
+	req.RequiredStatusChecks = rsc
 
 	rprr, err := expandRequiredPullRequestReviews(d)
 	if err != nil {
 		return nil, err
 	}
-	protectionRequest.RequiredPullRequestReviews = rprr
+	req.RequiredPullRequestReviews = rprr
 
 	res, err := expandRestrictions(d)
 	if err != nil {
 		return nil, err
 	}
-	protectionRequest.Restrictions = res
-
-	protectionRequest.EnforceAdmins = d.Get("enforce_admins").(bool)
+	req.Restrictions = res
 
-	return protectionRequest, nil
+	return req, nil
 }
 
 func flattenRequiredStatusChecks(d *schema.ResourceData, protection *github.Protection) error {
```

Comment:
shouldn't this be `flattenAndSetRequiredStatusChecks`

Reply:
Yeah, good point, I'll change that.